### PR TITLE
Send ping frame

### PR DESCRIPTION
**Pull request by plucury**
I'm using hyper to send messages to APNS, and it suggests checking the health of the connection with ping frame. However, I think it should be useful in other use cases.

### CODE REVIEW

File: test/test_hyper.py [hunk 1]
```diff
@@ -80,6 +80,20 @@ def test_connections_can_parse_ipv6_hosts_and_ports(self):
...
+
+        c = HTTP20Connection('www.google.com')
+        c._sock = DummySocket()
+        c._send_cb = data_callback
+        c.ping('00000000')
+
+        frames = list(frame_buffer)
+        assert len(frames) == 1
+        f = frames[0]
+        assert isinstance(f, PingFrame)
```
Comment (Lukasa) [inline test/test_hyper.py]: Can we enhance this check to confirm that the opaque data is passed through?

**Comment by Lukasa** [comment]
Thanks for this! This change generally looks good to me, though I have left one inline note.

**Comment by plucury** [comment]
@Lukasa Thanks!  Just updated my code.

**Comment by Lukasa** [comment]
Fab! I'm currently sat on a plane which means I can't effectively do the extra steps I like to do when I merge a PR, so if you don't mind I'll merge this a bit later in the day when I'm no longer awkwardly sat on an aircraft (or this flight might have wifi, in which case I'll do it here when this plane eventually takes off!)

**Comment by plucury** [comment]
@Lukasa That's fine. Have a nice trip!

**Comment by Lukasa** [comment]
Thanks @plucury! :sparkles: :cake: :sparkles:

**Pull request merged**